dandischema/utils.py

```diff
@@ -114,3 +114,25 @@ def strip_top_level_optional(type_: Any) -> Any:
     else:
         # `type_` is not an Optional
         return type_
+
+
+def sanitize_value(value: str, field: str = "non-extension", sub: str = "-") -> str:
+    """Replace all "non-compliant" characters with -
+
+    Of particular importance is _ which we use, as in BIDS, to separate
+    _key-value entries.  It is not sanitizing to BIDS level of clarity though.
+    In BIDS only alphanumerics are allowed, and here we only replace some known
+    to be offending symbols with `sub`.
+
+    When `field` is not "extension", we also replace ".".
+
+    Based on dandi.organize._sanitize_value.
+
+    .. versionchanged:: 0.8.3
+
+            ``sanitize_value`` added
+    """
+    value = re.sub(r"[_*\\/<>:|\"'?%@;]", sub, value)
```

Comment:
in dandi-cli ATM

```
dandi/organize.py=def _sanitize_value(value, field):
dandi/organize.py:    value = re.sub(r"[_*\\/<>:|\"'?%@;,\s]", "-", value)
```

so let's add those new (? added in Nov 2023 in 8d7b1a6492a8e51875285f89afce08fd2911281d so after initiated this PR)

```suggestion
    value = re.sub(r"[_*\\/<>:|\"'?%@;,\s]", sub, value)
```